### ISSUE TITLE
How query by objectID

### ISSUE THREAD

**Issue by Kikobeats** [algolia/instantsearch.js#2021]
![](https://d3uepj124s5rcx.cloudfront.net/items/0n200o460Q2A0Y1j030I/Screen%20Recording%202017-03-01%20at%2009.34%20pm.gif?v=236e3467)

Hi,

One of the main user case for use instantsearch is for ecommerce. 

Under this scenario, instantsearch works really neat for get real time feedback. 

But this is one of a serie of steps to do after buy something. Normally, the next step is know more information about the item, something like, "hey I'm interested in this, tell me more".

This issue is related with this scenario. 

How to design a one product view for individual items?

we need to contemplate two scenarios variations for that:

- You are navigating in the App and decide know more about a item.
- You have a external link related with a particular item.

Also, we need a unique value. We are going to use `objectID` associated with each item in the index.

In the first case, when you are navigating in the app and decide follow a item, in this scenario, actually you have the item. Then you can control if show or hide the item in a dialog, as the Twitter example, an generate the absolute url.

The second case have a different precondition but basically the same. You need to define a `route` for fetch the item, something like 

```jsx
<Router history={browserHistory}>
        <Route path='/' component={component} />
        <Route path='/:objectID' component={component} />
      </Router>
```

And a way to connect the route with `InstantSearch` to say that the `objectID` is the criteria for get the item.

---

What do you think, guys? 
Could be possible with the current basecode, or need a extra component for that?

Also I think that this is a good approach for improve the SEO of a instantsearch application: Could be possible define a worker an create a `sitemap.xml` using all the `objectIDs` of the index.

Related: https://github.com/DeuxHuitHuit/algolia-webcrawler

**Comment by bobylito** [comment]
@Kikobeats Thanks for your issue. It goes beyond the scope that we currently envision for instantsearch (react or vanilla). That being said, it's true that the records could contain all the necessary informations to be able to display a detail page. I think we should sync up on the topic with the core team but that's definitely an interesting direction. I think we'll have a first idea on what we're gonna do on that topic next week :)

**Comment by Kikobeats** [comment]
I tested something: Currently is possible query by ObjectID

<img width="603" alt="screen shot 2017-03-13 at 23 57 32" src="https://cloud.githubusercontent.com/assets/2096101/23878634/db1dd3a4-0848-11e7-8879-b5daa9327701.png">

This picture is using index panel and I get the result.

So basically, Need to do the same but from `<InstantSearch/>`

I saw that the component serialize the widgets urls. 

For example, if I have a `condition` refinement with `used|new` as value, the url for filter by `new` looks like:

```
https://market.windtoday.co/?refinementList[condition][0]=new
```

serialized into:

```
https%3A%2F%2Fmarket.windtoday.co%2F%3FrefinementList%5Bcondition%5D%5B0%5D%3Dnew
```

In the objectID case, I suppose that is necessary define the way to serialize the objectID value as the backend is expecting it?
